core/dbt/task/list.py

```diff
@@ -113,6 +113,8 @@ def generate_selectors(self):
             elif node.resource_type == NodeType.Unit:
                 assert isinstance(node, UnitTestDefinition)
                 unit_test_selector = ".".join([node.package_name, node.name])
+                if node.version:
```

Comment:
This should be a direct comparison to `None` to ensure `version=0` is handled correctly.

Reply:
It might be worth implementing a property on UnitTestDefinition.. maybe `versioned_name`? and consolidate this logic here + in formatting in test.py: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/task/test.py#L87-L88